### DISCUSSION TITLE
How to add link to interactive node

### DISCUSSION THREAD

**Discussion by mfoitzik** [ueberdosis/tiptap#1985] [General Questions] [answered]
I built an editor with TipTap 2 and Quasar 2 and used an interactive node view to deal with image properties, resizing, etc. A demo can be viewed at:[ https://www.simplicityblocks.com/quasar2tiptap2]( https://www.simplicityblocks.com/quasar2tiptap2) and the repo is at [https://github.com/mfoitzik/quasar2-tiptap2](https://github.com/mfoitzik/quasar2-tiptap2). My question/issue is when I select the image and try to add a link, the link does not get added. Can anyone offer guidance or point me in the right direction as to how to get this to work? Secondarily IO noticed that in Firefox I cannot select the interactive node (image) or see the cursor (Chrome works fine). Thank you.

**Comment by philippkuehn** [comment]
A link (a mark) can only added to inline images. Maybe that’s the issue?

**Reply by mfoitzik** [reply]
My custom extension for this interactive node has the following:
```
group: 'inline',

  content: 'inline*',
  inline: true,
  draggable: true,
```
I am not sure if that is what you mean.
I have a workaround where I can add additional attributes inside the interactive node and then wrap the img with an anchor tag in renderHTML, but I was hoping to create a more "standard" way of adding links to an image instead of having it in my image properties dialog.

**Comment by mfoitzik** [comment, marked as answer]
I am adding my answer as to how I solved this just in case it helps someone else. I had two things I needed to do to solve this problem. The first thing was that inside the editor configuration I had to load the Image extension AFTER my custom extension:
```
        VueComponent,
        Image,
```
The second thing I had to do is inside my custom extension I had a click event handler for turning on the interface for manipulating the image (resizing, dragging, selecting a settings menu). Inside the click event handler I had to add the following to make sure the node was selected:
`editor.value?.commands.setNodeSelection(props.getPos())`
